synapse/handlers/account_validity.py

```diff
@@ -98,6 +99,19 @@ async def on_user_registration(self, user_id: str) -> None:
         for callback in self._module_api_callbacks.on_user_registration_callbacks:
             await callback(user_id)
 
+    async def on_user_login(
+        self, user_id: str, auth_provider_type: LoginType, auth_provider_id: str
```

Comment:
Unfortunately `LoginType` is not an enum:

```suggestion
        self, user_id: str, auth_provider_type: str, auth_provider_id: str
```

Reply:
Fixed!